CMakeLists.txt

```diff
@@ -133,6 +134,10 @@ if(CMAKE_BUILD_TYPE MATCHES Release)
   endif()
 endif()
 
+if (COVERAGE)
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -ftest-coverage")
```

Comment:
Do we need `-g -O0` or at least `-fno-inline` in order for this to work properly?